exotic/fit_centroid_old.py

```diff
@@ -0,0 +1,600 @@
+import numpy as np
+from dataclasses import dataclass
+
+from scipy.optimize import least_squares
+from scipy.ndimage import median_filter
+from scipy.interpolate import RectBivariateSpline
+
+from photutils import aperture_photometry
+from photutils import CircularAperture
+
+from astropy.io import fits
+
+from skimage.registration import phase_cross_correlation
+
+try:
+    from api.elca import lc_fitter, binner, transit
+except ImportError:
+    from .api.elca import lc_fitter, binner, transit
+
+
+def mesh_box_prev(pos, box):
+    pos = [int(np.round(pos[0])), int(np.round(pos[1]))]
+    x = np.arange(pos[0] - box, pos[0] + box + 1)
+    y = np.arange(pos[1] - box, pos[1] + box + 1)
+    xv, yv = np.meshgrid(x, y)
+    return xv.astype(int), yv.astype(int)
+
+
+# Method fits a 2D gaussian function that matches the star_psf to the star image and returns its pixel coordinates
+def fit_centroid_prev(data, pos, init=None, lossfn='linear', box=10):
+    if init is None:  # if init is none, then set the values
+        init = [-1, 5, 5, 0]
+    else:  # remove rotation
+        init = np.delete(init, 3)
+
+    # estimate the amplitude and centroid
+    if init[0] == -1:
+        # subarray of data around star
+        xv, yv = mesh_box_prev(pos, box)
+
+        # amplitude guess
+        init[0] = np.max(data[yv, xv])
+
+        # weighted sum to estimate center
+        wx = np.sum(np.unique(xv) * data[yv, xv].sum(0)) / np.sum(data[yv, xv].sum(0))
+        wy = np.sum(np.unique(yv) * data[yv, xv].sum(1)) / np.sum(data[yv, xv].sum(1))
+        pos = [wx, wy]
+        # estimate std by calculation of FWHM
+        x, y = data[yv, xv].sum(0), data[yv, xv].sum(1)
+        init[1] = estimate_sigma_prev(x)
+        init[2] = estimate_sigma_prev(y)
+
+        # Background Estimate
+        # compute the average from 1/4 of the lowest values in the background
+        init[3] = np.mean(np.sort(data[yv, xv].flatten())[:int(data[yv, xv].flatten().shape[0] * 0.25)])
+    # print('init priors for centroid:',init)
+    # print('init2:',init)
+
+    # recenter data on weighted average of light (peak amplitude)
+    xv, yv = mesh_box_prev(pos, box)
+
+    # pars = x,y, a,sigx,sigy, rotate
+    def fcn2min(pars):
+        model = star_psf_prev(xv, yv, *pars)
+        return (data[yv, xv] - model).flatten()  # method for LS
+        # return np.sum( (data[yv,xv]-model)**2 ) # method for minimize
+
+    lo = [pos[0] - box, pos[1] - box, 0, 1, 1, 0]
+    up = [pos[0] + box, pos[1] + box, 100000, 40, 40, np.max(data[yv, xv])]
+    res = least_squares(fcn2min, x0=[*pos, *init], bounds=[lo, up], loss=lossfn, jac='3-point')
```

Comment:
I'm going to assume it's coming from the least squares method. I've played around with the mesh box and gaussian psf from both the older version and newer, yet both yield the same error.